async/xen_api_async_unix.ml

```diff
@@ -32,18 +32,16 @@ module IO = struct
   let iter fn x = Deferred.List.iter x ~f:fn
 
   let read_line (_, ic) =
-    Reader.read_line ic >>=
-    function
-    |`Ok s -> return (Some s)
-    |`Eof -> return None
+    Reader.read_line ic
+    >>| function
+    | `Ok s -> Some s
+    | `Eof -> None
 
-  let read =
-    let buf = String.create 4096 in
-    fun (_, ic) len ->
-      Reader.read ic ~len buf >>=
-      function
-      |`Ok len' -> return (String.sub buf 0 len')
-      |`Eof -> return ""
+  let read (_, ic) len =
+    let buf = Bytes.create len in
```

Comment:
So this means that the buffer `buf` won't be shared any more between subsequent invocations of `read`, but a new one will be created each time we call it - is this the change that fixed the Async bindings?

Reply:
Yes